File: Task3/javaScript/main.js
// Функция для суммы цифр числа
function sumOfDigits(number) {
    return number
        .toString()
        .split('')
        .reduce((sum, digit) => sum + Number(digit), 0);
}

// Функция для произведения цифр числа
function productOfDigits(number) {
    return number
        .toString()
        .split('')
        .reduce((prod, digit) => prod * Number(digit), 1);
}

// Чтение N и массива чисел
const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
});

readline.question('Введите количество чисел N: ', nInput => {
    const N = parseInt(nInput);

    readline.question(`Введите ${N} чисел: `, line => {
        const numbers = line.split(' ').map(Number);

        if (numbers.length !== N) {
            console.log('Количество введённых чисел не соответствует N');
            readline.close();
            return;
        }

        // Фильтруем числа по условию
        const result = numbers.filter(num => sumOfDigits(num) < productOfDigits(num));

        // Вывод результата
        console.log('Результат: ');
        if (result.length === 0) {
            console.log('не найдено');
        } else {
            console.log(result.join(' '));
        }

        readline.close();
    });
});

File: Task3/swift/main.swift
import Foundation

// Функция для вычисления суммы цифр числа
func sumOfDigits(_ number: Int) -> Int {
    return String(number).compactMap { $0.wholeNumberValue }.reduce(0, +)
}

// Функция для вычисления произведения цифр числа
func productOfDigits(_ number: Int) -> Int {
    return String(number).compactMap { $0.wholeNumberValue }.reduce(1, *)
}

// Чтение числа N
print("Введите количество чисел N:")
guard let inputN = readLine(), let N = Int(inputN) else {
    fatalError("Некорректный ввод N")
}

// Чтение последовательности чисел
print("Введите \(N) чисел:")
guard let line = readLine() else {
    fatalError("Ошибка ввода чисел")
}

let numbers = line.split(separator: " ").compactMap { Int($0) }

if numbers.count != N {
    fatalError("Количество введённых чисел не соответствует N")
}

// Фильтруем числа, у которых сумма цифр < произведение цифр
let result = numbers.filter { sumOfDigits($0) < productOfDigits($0) }

// Вывод результата
print("Результат: ")
if result.isEmpty {
    print("не найдено")
} else {
    print(result.map { String($0) }.joined(separator: " "))
}